File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/util/ObjectToRawJsonStringConverter.java
package com.attentive.androidsdk.internal.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.util.StdConverter;
import timber.log.Timber;


public class ObjectToRawJsonStringConverter extends StdConverter<Object, String> {
    private final ObjectMapper objectMapper = new ObjectMapper();

    @Override
    public String convert(Object value) {
        try {
            return objectMapper.writeValueAsString(value);
        } catch (JsonProcessingException e) {
            Timber.e(e, "Failed to convert object to JSON string");
            throw new RuntimeException(e);
        }
    }
}

File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/util/ObjectToRawJsonStringConverter.kt
package com.attentive.androidsdk.internal.util

import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import timber.log.Timber

class ObjectToRawJsonStringConverter {

    fun convert(value: Any?): String {
        return try {
            Json.encodeToString(value)
        } catch (e: Exception) {
            Timber.e(e, "Failed to convert object to JSON string")
            throw RuntimeException(e)
        }
    }
}